src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+
+using Dynamo.Graph;
+using Dynamo.Graph.Nodes;
+
+using ProtoCore.AST.AssociativeAST;
+using CoreNodeModels.Properties;
+
+using DSColor = DSCore.Color;
+
+using Autodesk.DesignScript.Runtime;
+
+
+namespace CoreNodeModels.Input
+{
+    [NodeName("Color Palette")]
+    [NodeDescription("ColorPaletteDescription", typeof(Resources))]
+    [NodeCategory("Core.Color.Create")]
+    [NodeSearchTags("ColorUISearchTags", typeof(Resources))]
+    [IsDesignScriptCompatible]
+    [OutPortNames("Color")]
+    [OutPortTypes("Color")]
+    [OutPortDescriptions("Selected Color.")]
+    public class ColorPalette : NodeModel
+    {
+        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
+        public DSColor dsColor
+        {
+            get { return dscolor; }
+            set
+            {                              
+                dscolor = value;
+                OnNodeModified();
+            }
+        }
+        /// <summary>
+        ///     Node constructor.
+        /// </summary>
+        public ColorPalette()
+        {
+            RegisterAllPorts();
+        }
+        private DSColor DeserializeValue(string val)
+         {
+            try
+            {
+                var nums = new List<int>();
```

Comment:
can you send the information you have on regex being an issue on mono - I have not seen that before and if you need to do this workaround it will need comments inline.... let's make sure we can't just use a regex though.